### ISSUE TITLE
[BUG]: the RegExp replacement should text supports escape characters, such as \n

### ISSUE THREAD

**Issue by miaogaolin** [ObsidianPublisher/obsidian-github-publisher#254]
### Issue validation

- [X] I checked the issue to prevent duplicate
- [X] My issue concern doesn't concern the [Mkdocs template](https://github.com/ObsidianPublisher/template-main/)

## Plugin Settings 
![image](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/57740293/58d205a8-f96c-4a4f-9f83-0267f25704e0)

## Expectation:
Before Replacing:
```
cover.image
```
Aftere Replacing
```
cover:
  image
```
## But the actual result:
```
cover:\n\timage
```

### OS

Windows

### Plugin version

6.11.5

### Obsidian version & debug log

```bash session
Obsidian version: 1.4.16
```

**Comment by Lisandra-dev** [comment]
Well...
For this, I don't know why.
I will investigate why it replace by the strict string and not the "value"

**Comment by miaogaolin** [comment]
Thank you so much.

**Comment by Lisandra-dev** [comment]
Fixed by de346721bf147c8e0bb0112627adb91b9402e388
